compiler/src/compile.rs

```diff
@@ -1874,10 +1879,10 @@ impl<O: OutputStream> Compiler<O> {
         assert!(scope.sub_scopes.is_empty());
     }
 
-    fn lookup_name(&self, name: &str) -> &Symbol {
+    fn lookup_name(&self, name: &str) -> Option<&Symbol> {
         // println!("Looking up {:?}", name);
         let scope = self.scope_stack.last().unwrap();
-        scope.lookup(name).unwrap()
```

Comment:
The failure of this unwrap is caused elsewhere. The symbol `void` must be present in a symbol table. It might be better to modify `symboltable.rs` to fix this particular issue.

My suggestion:
- change the `unwrap` call into an `except("The symbol must be present in the symbol table, even when it is undefined in python.")` call. In other words, the unwrap here must succeed, otherwise an internal error was made, and we should not leak internal rustpython errors into python runtime errors, but fail with an assertion.
- Move the lines concering the return type annotation (https://github.com/RustPython/RustPython/blob/master/compiler/src/symboltable.rs#L251) before the `enter_function` call. This might also fix the issue.

If anything is unclear, please let me know.